edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/validations/ShippingInstructionValidator.java

```diff
@@ -149,6 +153,51 @@ private void validateShipper(ValidationState<ShippingInstruction> state) {
     }
   }
 
+  private void validateManifestFilings(ValidationState<ShippingInstruction> state) {
+
+    List<AdvanceManifestFilingEBL> advanceManifestFilingSIs = state.getValue().getAdvanceManifestFilings();
+    List<ConsignmentItem> consignmentItems = state.getValue().getConsignmentItems();
+
+    ConsignmentItem firstconsignmentItem = state.getValue().getConsignmentItems().get(0);
+    List<AdvanceManifestFiling> advanceManifestFilingBase = firstconsignmentItem.getShipment().getAdvanceManifestFilings();
+    List<AdvanceManifestFiling> misMatchedManifestFilings = new ArrayList<>();
+
+    for(ConsignmentItem ci: consignmentItems) {
+      List<AdvanceManifestFiling> advanceManifestFilings = ci.getShipment().getAdvanceManifestFilings();
+      misMatchedManifestFilings = advanceManifestFilingBase.stream()
+        .filter(two -> advanceManifestFilings.stream()
+          .noneMatch(one -> one.getManifestTypeCode().equals(two.getManifestTypeCode())
+            && two.getCountryCode().equals(one.getCountryCode())))
+        .toList();
+    }
```

Comment:
Doesn't this "overwrite" the misMatchedManifestFilings for every run? If last run succeeds - then the **if** statement below will not run...